src/vanilla/utils/atomWithStorage.ts

```diff
@@ -141,30 +192,18 @@ export function createJSONStorage<Value>(
       ),
     removeItem: (key) => getStringStorage()?.removeItem(key),
   }
+
   if (
     typeof window !== 'undefined' &&
-    typeof window.addEventListener === 'function' &&
-    window.Storage
+    typeof window.addEventListener === 'function'
   ) {
-    storage.subscribe = (key, callback, initialValue) => {
-      if (!(getStringStorage() instanceof window.Storage)) {
-        return () => {}
-      }
-      const storageEventCallback = (e: StorageEvent) => {
-        if (e.storageArea === getStringStorage() && e.key === key) {
-          let newValue: Value
-          try {
-            newValue = JSON.parse(e.newValue || '')
-          } catch {
-            newValue = initialValue
-          }
-          callback(newValue)
-        }
-      }
-      window.addEventListener('storage', storageEventCallback)
-      return () => {
-        window.removeEventListener('storage', storageEventCallback)
-      }
+    if (getStringStorage()?.subscribe) {
```

Comment:
AFAIR, this can throw. So, we can't call it here.

Reply:
I believe the `removeEventListener` method won't throw unless the given parameters are wrong which is not happening in our case. am I wrong?

and this code is the same as before:

Reply:
Oh, my bad. I was thinking of old implementation. We catch it already.

Reply:
My first comment was actually right. We can't call `getStringStorage()` on creation. #2581